nexus/src/app/sagas/instance_start.rs

```diff
@@ -77,6 +88,15 @@ impl NexusSaga for SagaInstanceStart {
         _params: &Self::Params,
         mut builder: steno::DagBuilder,
     ) -> Result<steno::Dag, SagaInitError> {
+        builder.append(Node::action(
+            "propolis_id",
+            "GeneratePropolisId",
+            ACTION_GENERATE_ID.as_ref(),
+        ));
+
+        builder.append(alloc_server_action());
```

Comment:
(Note: I do not want to change this now, but asking out of curiosity)

From your "Known Issues" in the PR message:

> The virtual provisioning table is still updated only at instance creation/deletion time. Usage metrics that depend on this table might report strange and wonderful values if a user creates many more instances than can be started at one time.

We could basically "just" move the virtual provisioning saga step from instance creation here (and the corresponding release from "instance delete" to "instance stop"), right? Do you foresee anything being particularly more complex about this, after this PR merges?

I'm not even sure we **should** do so, but asking mostly to understand what we could do.

Reply:
Follow-up -- I see some comments below about safely unwinding this saga if it is called concurrently with distinct propolis IDs. I see that would probably also be an additional refactor necessary for correctness -- the "propolis ID"s would need to consume virtual resources, rather than instance IDs, otherwise unwinding them wouldn't be safe.

Reply:
Is this the same problem as we had in #2839? Seems like it wouldn't be too hard to resolve by a similar method if we wanted (use Propolis IDs instead of instance IDs, maybe change an enum variant or two around to reflect the sort of thing that's reserving the resources).

Reply:
Filed #4257 to track this.